version: "3.1"

services:
  db:
    image: postgres:10
    ports:
      # We changed the default port so people having already a postgres instance running on port 5432 won't have a port conflict
      - "5433:5432"
    volumes:
      - ./postgres-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_PASSWORD=password
      - POSTGRES_USER=postgres
      - POSTGRES_DB=postgres

  backend:
    build: ./backend
    command: python /code/manage.py rundebugserver 0.0.0.0:80
    volumes:
      - ./backend:/code
    ports:
      - "8000:80"
      - "9000:9000"
    links:
      - db
      - redis
    env_file:
      - .env
    environment:
      DATABASE_URL: "postgres://postgres:password@db:5432/postgres"

  redis:
    image: redis

  celery:
    build: ./backend
    volumes:
      - ./backend:/code
    links:
      - db
      - redis
    env_file:
      - .env
    environment:
      DATABASE_URL: "postgres://postgres:password@db:5432/postgres"
    command: celery worker -A root.celery -B -l INFO --scheduler django_celery_beat.schedulers:DatabaseScheduler -c 4

networks:
  default:
    ipam:
      config:
        - subnet: 172.16.210.0/24
